api/provider/aws/service_read.go

```diff
@@ -9,17 +9,23 @@ import (
 )
 
 func (s *ServiceProvider) Read(serviceID string) (*models.Service, error) {
-	model := &models.Service{}
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return nil, err
+	}
 
-	clusterName := addLayer0Prefix(s.Config.Instance(), model.EnvironmentID)
+	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), environmentID)
+	clusterName := fqEnvironmentID
 	service, err := s.readService(clusterName, serviceID)
 	if err != nil {
 		return nil, err
 	}
 
-	model.DesiredCount = aws.Int64Value(service.DesiredCount)
-	model.RunningCount = aws.Int64Value(service.RunningCount)
-	model.PendingCount = aws.Int64Value(service.PendingCount)
+	model := &models.Service{
+		DesiredCount: aws.Int64Value(service.DesiredCount),
+		RunningCount: aws.Int64Value(service.RunningCount),
+		PendingCount: aws.Int64Value(service.PendingCount),
+	}
```

Comment:
[line 48] please use the standard name in regards to  ` s.updateWithTagInfo` -> `populateModelTags`

Reply:
that helper function should exist in this file. Also, `ListTags()` should not be using that function.  It should have a special function that is optimized for list instead of read (current implementation makes N calls to the db. Optimized one can make only 2 or 3).